core/src/main/java/com/datastax/oss/driver/internal/core/channel/ChannelFactory.java

```diff
@@ -161,12 +162,20 @@ private void connect(
       CompletableFuture<DriverChannel> resultFuture) {
 
     NettyOptions nettyOptions = context.getNettyOptions();
+    Duration connectTimeout =
+        context
+            .getConfig()
+            .getDefaultProfile()
+            .getDuration(DefaultDriverOption.CONNECTION_CONNECT_TIMEOUT);
 
     Bootstrap bootstrap =
         new Bootstrap()
             .group(nettyOptions.ioEventLoopGroup())
             .channel(nettyOptions.channelClass())
             .option(ChannelOption.ALLOCATOR, nettyOptions.allocator())
+            .option(
+                ChannelOption.CONNECT_TIMEOUT_MILLIS,
+                Long.valueOf(connectTimeout.toMillis()).intValue())
```

Comment:
@olim7t Should I move this bit into `afterBootstrapInitialized()`?

Reply:
Yes, let's move it there for consistency, that's where the other socket options are processed.